src/modules/Footer.jsx

```diff
@@ -0,0 +1,269 @@
+  const current = history[history.length - 1];
+  const { board, score, win, gameOver } = current;
+  const squareBoardSize = boardSize * boardSize;
+  let newScore = 0;
+  const [play, setPlay] = useState(false);
+
+  const isWin = (array) => {
+    const tempArr = [...array].sort((a, b) => a - b);
+    return tempArr[tempArr.length - 1] >= winningNumber;
+  };
+
+  const handleGameOver = () => {
+    if (!gameOver) {
+      setOpenLose(true);
+      setPlay(false);
+    }
+
+    return true;
+  };
+
+  const setNewBoard = (newBoard) => {
+    const checkObj = {
+      resultBoard: null,
+      winCheck: null,
+      gameOver: null,
+    };
+
+    if (!areArraysEqual(newBoard, board)) {
+      const tempBoard = getNewBoardArray(newBoard);
+      checkObj.resultBoard = tempBoard !== -1 ? tempBoard : newBoard;
+      if (isWin(checkObj.resultBoard) && !win) {
+        checkObj.winCheck = true;
+        setOpenWin(true);
+      }
+      // eslint-disable-next-line no-use-before-define
+      if (isGameOver(checkObj.resultBoard)) checkObj.gameOver = handleGameOver();
+      if (!gameOver) {
+        setHistory(
+          history.concat({
+            board: checkObj.resultBoard || board,
+            score: score + newScore,
+            win: checkObj.winCheck || win,
+            gameOver: checkObj.gameOver || gameOver,
+          })
+        );
+
+        setValueInLocalStorage(
+          '2048-history',
+          history.concat({
+            board: checkObj.resultBoard || board,
+            score: score + newScore,
+            win: checkObj.winCheck || win,
+            gameOver: checkObj.gameOver || gameOver,
+          })
+        );
+      }
+    }
+    const newBestScore = score + newScore > currentBestScore ? score + newScore : null;
+    if (newBestScore) {
+      setBestScore(bestScore.concat(newBestScore));
+      setValueInLocalStorage('bestScore', bestScore.concat(newBestScore));
+    }
+  };
+
+  const processLine = (line, direction = false) => {
+    const trimLine = line.filter((elem) => !!elem);
+    if (direction) trimLine.reverse();
+    for (let j = 0; j < trimLine.length; j++) {
+      if (trimLine[j] === trimLine[j + 1]) {
+        newScore += trimLine[j] + trimLine[j + 1];
+        [trimLine[j], trimLine[j + 1]] = [trimLine[j] + trimLine[j + 1], 0];
+      }
+    }
+    const newLine = trimLine.filter((elem) => !!elem);
+    while (newLine.length < line.length) {
+      newLine.push(0);
+    }
+    return direction ? newLine.reverse() : newLine;
+  };
+
+  const getRow = (array, i) => array.filter((elem, id) => id >= i && id < i + boardSize);
+
+  const getColumn = (array, i) => array.filter((elem, id) => (id - i) % boardSize === 0);
+
+  const updateNewBoardHorizontal = (newBoard, i, direction = false) => {
+    const newLine = processLine(getRow(board, i), direction);
+    newLine.map((elem, id) => {
+      // eslint-disable-next-line no-param-reassign
```

Comment:
you can copy the argument to a variable and work with it if you need

Reply:
done, don't know why I haven't done it at the time instead of turning off lint